src/schedule/edf_schedule.c

```diff
@@ -295,8 +295,6 @@ static void scheduler_free_edf(void *data)
 	/* free main task context */
 	task_main_free();
 
-	list_item_del(&edf_sch->list);
```

Comment:
yes, like in https://github.com/thesofproject/sof/commit/7bdbe295ef0cbafdd047d5352fb343c6f33fe2d0 the author really actually wanted to just initialise the list head... But yes, this is called when the core powers down and after that the complete heap will be freed anyway. And besides at this moment the list should be empty anyway, i.e. also already initialised. So, you could in fact remove `list_init()` from `scheduler_free_ll()` too.

Reply:
fixed that too